### PR TITLE
Wearing flowers now gives you a mood boost + small flower code refactor

### PR DESCRIPTION
**Pull request by CydiaLamiales**
## About The Pull Request
Wearing flowers and flower crowns now gives you a +1 and +3 mood boosts respectively.

## Why It's Good For The Game
Allows Botany to create more meaningful items to the crew.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

image

image

Heretic ritual accepting a forget-me-not

image

image

All the flowers grown/mutated as normal

image

</details>

## Changelog
:cl:
add: Added new positive moodlets for wearing flowers and flower crowns
refactor: repathed grown flowers into /obj/item/reagent_containers/food/snacks/grown/flower/
refactor: repathed flower seeds into /obj/item/seeds/flower/
spellcheck: fixed a typo in the sunflower attack text
tweak: Changed the heretic ritual text to reflect it accepting any flower, made them accept Harebells and Rainbow Bunch flowers
/:cl:

### CODE REVIEW

**Comment by CydiaLamiales** [comment]
Youwhat.

**Comment by CydiaLamiales** [comment]
There we go, no more merge conflicts.

**Comment by CydiaLamiales** [comment]
Actually, nvm, I can do some more work on this.

**Comment by CydiaLamiales** [comment]
Alright that's about everything.
image
image
Isn't this much better to look at and navigate.

**Review by imsynthex — APPROVED** [review]
Code looks solid 👍🏼

File: code/datums/components/crafting/tailoring.dm [hunk 1]
```diff
@@ -312,32 +312,32 @@
 
 /datum/crafting_recipe/rainbowbunchcrown
 	name = "Rainbow Flower Crown"
-	result = /obj/item/clothing/head/rainbowbunchcrown/
+	result = /obj/item/clothing/head/flowercrown/rainbowbunch/
```
Comment (itsmeow) [inline code/datums/components/crafting/tailoring.dm]: favor, please remove trailing `/` on all these result things. they're very bad

Comment (CydiaLamiales) [inline code/datums/components/crafting/tailoring.dm]:
```suggestion
	result = /obj/item/clothing/head/flowercrown/rainbowbunch
```

File: code/datums/components/crafting/tailoring.dm [hunk 2]
```diff
@@ -312,32 +312,32 @@
...
 	time = 20
-	reqs = list(/obj/item/reagent_containers/food/snacks/grown/rainbow_flower = 5,
+	reqs = list(/obj/item/reagent_containers/food/snacks/grown/flower/rainbow = 5,
 				/obj/item/stack/cable_coil = 3)
 	category = CAT_CLOTHING
 
 /datum/crafting_recipe/sunflowercrown
 	name = "Sunflower Crown"
-	result = /obj/item/clothing/head/sunflowercrown/
+	result = /obj/item/clothing/head/flowercrown/sunflower/
```
Comment (CydiaLamiales) [inline code/datums/components/crafting/tailoring.dm]:
```suggestion
	result = /obj/item/clothing/head/flowercrown/sunflower
```

File: code/datums/components/crafting/tailoring.dm [hunk 3]
```diff
@@ -312,32 +312,32 @@
...
 	time = 20
 	reqs = list(/obj/item/grown/sunflower = 5,
 				/obj/item/stack/cable_coil = 3)
 	category = CAT_CLOTHING
 
 /datum/crafting_recipe/poppycrown
 	name = "Poppy Crown"
-	result = /obj/item/clothing/head/poppycrown/
+	result = /obj/item/clothing/head/flowercrown/poppy/
```
Comment (CydiaLamiales) [inline code/datums/components/crafting/tailoring.dm]:
```suggestion
	result = /obj/item/clothing/head/flowercrown/poppy
```

File: code/datums/components/crafting/tailoring.dm [hunk 4]
```diff
@@ -312,32 +312,32 @@
...
 	time = 20
-	reqs = list(/obj/item/reagent_containers/food/snacks/grown/poppy = 5,
+	reqs = list(/obj/item/reagent_containers/food/snacks/grown/flower/poppy = 5,
 				/obj/item/stack/cable_coil = 3)
 	category = CAT_CLOTHING
 
 /datum/crafting_recipe/lilycrown
 	name = "Lily Crown"
-	result = /obj/item/clothing/head/lilycrown/
+	result = /obj/item/clothing/head/flowercrown/lily/
```
Comment (CydiaLamiales) [inline code/datums/components/crafting/tailoring.dm]:
```suggestion
	result = /obj/item/clothing/head/flowercrown/lily
```

**Review by CydiaLamiales — COMMENTED** [review]
Removing trailing lines

File: code/datums/components/crafting/tailoring.dm (hunk 1 above)
Comment (CydiaLamiales) [reply code/datums/components/crafting/tailoring.dm]: There we go, removed.

**Review by itsmeow — APPROVED** [review]
flower

**Review by itsmeow — APPROVED** [review]

**Comment by itsmeow** [comment]
Aw man another conflict

**Comment by CydiaLamiales** [comment]
Fucking finally, I hate map merge conflicts.

**Review by itsmeow — APPROVED** [review]

**Pull request merged**